components/builder-api-client/src/builder.rs

```diff
@@ -186,7 +207,13 @@ impl BuilderAPIClient {
         fs::create_dir_all(&dst_path)?;
         let file_name = response::get_header(&resp, X_FILENAME)?;
         let dst_file_path = dst_path.join(file_name);
-        let w = AtomicWriter::new(&dst_file_path)?;
+        let w = {
+            let mut w = AtomicWriter::new(&dst_file_path)?;
+            if let Permissions::Explicit(permissions) = permissions {
```

Comment:
> To make further distinctions would be tricky and a bit more complicated.

Is this where you would make that distinction? Essentially, only add the explicit permissions if it is the `/hab/cache/artifacts` directory.

Reply:
It's not that it's impossible, it's just that it would be a bit messy and would end up distributing filesystem awareness in various parts of the codebase.